beacon-chain/sync/initial-sync/blocks_fetcher_utils.go

```diff
@@ -286,17 +286,17 @@ func (f *blocksFetcher) findAncestor(ctx context.Context, pid peer.ID, block *et
 }
 
 // bestFinalizedSlot returns the highest finalized slot of the majority of connected peers.
-func (f *blocksFetcher) bestFinalizedSlot() uint64 {
+func (f *blocksFetcher) bestFinalizedSlot() types.Slot {
 	finalizedEpoch, _ := f.p2p.Peers().BestFinalized(
 		params.BeaconConfig().MaxPeersToSync, f.chain.FinalizedCheckpt().Epoch)
-	return uint64(finalizedEpoch) * params.BeaconConfig().SlotsPerEpoch
+	return params.BeaconConfig().SlotsPerEpoch.Mul(uint64(finalizedEpoch))
 }
 
 // bestNonFinalizedSlot returns the highest non-finalized slot of enough number of connected peers.
-func (f *blocksFetcher) bestNonFinalizedSlot() uint64 {
+func (f *blocksFetcher) bestNonFinalizedSlot() types.Slot {
 	headEpoch := helpers.SlotToEpoch(f.chain.HeadSlot())
 	targetEpoch, _ := f.p2p.Peers().BestNonFinalized(flags.Get().MinimumSyncPeers*2, headEpoch)
-	return uint64(targetEpoch) * params.BeaconConfig().SlotsPerEpoch
+	return params.BeaconConfig().SlotsPerEpoch.Mul(uint64(targetEpoch))
```

Comment:
Do you mean revert, like in making `bestNonFinalizedSlot()` return `uint64` (instead of `types.Slot`), or reverting just the calculation, that's changing the line about to
```golang
return types.Slot(uint64(targetEpoch) * uint64(params.BeaconConfig().SlotsPerEpoch))
```
When you are worried about the panic, you mean overflow (if `targetEpoch` is huge)? Aren't we better off with using `SafeMul()` in that case and handling that overflow?

Reply:
Ah no, I was mistaking it with something else. This should be fine.